backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -468,7 +468,7 @@ export class ExperimentAssignmentService {
       ]);
     } else {
       [experiments, [userExcluded, groupExcluded]] = await Promise.all([
-        this.experimentRepository.getValidExperiments(context),
+        this.experimentService.getCachedValidExperiments(context),
         this.checkUserOrGroupIsGloballyExcluded(experimentUser),
```

Comment:
`this.checkUserOrGroupIsGloballyExcluded(experimentUser)`
Can you move this before fetching the valid experiment and end the request if the user is excluded? And because this is cached doing query serially won't impact the performance.